### ISSUE TITLE
False-Positive (reportUnsafeMultipleInheritance) when using ABC or a Protocol with pydantic BaseModel

### ISSUE THREAD

**Issue by Amolition** [DetachHead/basedpyright#1817]
## Description

When creating a subclass from `pydantic.BaseModel` with `abc.ABC` or `typing.Protocol` basedpyright reports this error: 

```
`reportUnsafeMultipleInheritance`:   Multiple inheritance is not allowed because the following base classes contain `__init__` or `__new__` methods that may not get called: BaseModel
```

despite using `ABC` with `BaseModel` being explicitly supported according to [pydantic documentation](https://pydantic.dev/docs/validation/latest/concepts/models/#abstract-base-classes). Even though `Protocol` isn't specifically mentioned I feel like this should probably also work since it should just define structural typing without affecting the behaviour of the derived class (besides some default attributes perhaps).

I'm not sure if this issue is specific to Pydantic's BaseModel or more general but this is where I came across it.

## Minimum Reproducible Example

**Python Version**: *3.12.13*

### requirements.txt

```
pydantic==2.13.4
basedpyright==1.39.6
```

### example.py

```py
import pydantic as pyd
from abc import ABC
from typing import Protocol

class X(pyd.BaseModel, ABC): # false-positive here
    pass

class YABC(ABC):
    pass

class Y(pyd.BaseModel, YABC): # false-positive here
    pass

class ZProtocol(Protocol):
    pass

class Z(pyd.BaseModel, ZProtocol): # false-positive here
    pass
```

**Comment by DetachHead** [comment]
this is related to https://github.com/DetachHead/basedpyright/issues/868#issuecomment-2472205770, but i guess we should special case `ABC` and `Protocol` because they would never have a constructor

**Comment by Amolition** [comment]
oh yeah I see you're already aware of the wider problem! A special case would be handy I guess if a general fix to the problem isn't easily forthcoming?
